#include "mainwindow.h"
#include "ui_mainwindow.h"
#include"registrationwindows.h"
#include"administratorlogin.h"
#include"userview.h"
#include"user.h"
#include"administrator.h"


#include<QRegExp>
#include<QRegularExpressionValidator>
#include<QPushButton>
#include<QTime>
#include<QString>
#include<QFileInfo>
#include<QFile>
#include<QDataStream>
#include<QDir>
#include<QMessageBox>
#include<QDate>
#include<QTextStream>

#include<QInputDialog>
#include<QLineEdit>
#include<QDialogButtonBox>

#include<QDebug>
#include<QPainter>
#include<QPixmap>


MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this);

//ËÆæÁΩÆËÉåÊôØÂíåÊ†áÈ¢ò
    setWindowTitle(QString("Ê®°ÊãüÊäïÁ•®Â∞èÁ®ãÂ∫è"));//Â≠óÁ¨¶‰∏≤

    ui->enterBtn->resize(64,64);
    ui->enterBtn -> setStyleSheet("#enterBtn{border:0px ;}");
    ui->enterBtn -> setStyleSheet("#enterBtn{border-image:url(:/resource/dengluoff.png);}");
    connect(ui->enterBtn, &QPushButton::pressed ,[=](){
       ui->enterBtn -> setStyleSheet("#enterBtn{border-image:url(:/resource/dengluon.png);}");
    });
    connect(ui->enterBtn, &QPushButton::released, [=](){
        ui->enterBtn -> setStyleSheet("#enterBtn{border-image:url(:/resource/dengluoff.png);}");
    });

    ui->registeredBtn->resize(64,64);
    ui->registeredBtn -> setStyleSheet("#registeredBtn{border:0px ;}");
    ui->registeredBtn -> setStyleSheet("#registeredBtn{border-image:url(:/resource/zhuceoff.png);}");
    connect(ui->registeredBtn, &QPushButton::pressed ,[=](){
       ui->registeredBtn -> setStyleSheet("#registeredBtn{border-image:url(:/resource/zhuceon.png);}");
    });
    connect(ui->registeredBtn, &QPushButton::released, [=](){
        ui->registeredBtn -> setStyleSheet("#registeredBtn{border-image:url(:/resource/zhuceoff.png);}");
    });


   // this->setStyleSheet("#MainWindow{border-image:url(:/resource/Vote1.png);}");

//    QPixmap pix(300,300);
//    bool a =
//    pix.load(":/resource/Vote1.png");
//    qDebug () << a ;
//    ui->label->setGeometry(1,1,300,300);
//    ui->label->setPixmap(pix);



//------------------------------Ê£ÄÊµãÊòØÂê¶Â≠òÂú®ÂØπÂ∫îÊñá‰ª∂---------------------------//
    InitialCheck();


//ËÆæÁΩÆÂØÜÁ†ÅÂíåÊâãÊú∫Âè∑Á†ÅËæìÂÖ•Ê†ºÂºèÁöÑ
    QRegExp regx("1([0-9]+)");
    QRegExp regx1("([0-9]+)");
    QValidator *validator = new QRegExpValidator(regx,ui->phoneinput);
    QValidator *validator1 = new QRegExpValidator(regx1,ui->codeinput);
    ui->phoneinput->setValidator(validator);
    ui->codeinput->setValidator(validator1);
//ËÆæÁΩÆÂØÜÁ†ÅÂíåÊâãÊú∫Âè∑Á†ÅËæìÂÖ•Ê†ºÂºè


    registrationWindows *reg = new registrationWindows(this);  //ÂàõÂª∫‰∏Ä‰∏™Ê≥®ÂÜåÁïåÈù¢



//ËøôÈáåÊòØÁÇπÂáª‰∏ªÁïåÈù¢ÁöÑÊ≥®ÂÜåÊåâÈíÆÔºåÁÑ∂ÂêéËøõÂÖ•Ê≥®ÂÜåÁöÑÈ°µÈù¢
    connect(ui->registeredBtn,&QPushButton::clicked,[=](){
        reg->show();
        beauty();
        this->hide();
    });

    connect(reg , &registrationWindows::registrationtomain,[=](){

        this->show();
        beauty();
        reg->close();
        reg->clearLineEdit();
        });//Â§ÑÁêÜÂáΩÊï∞
//ËøôÈáåÊòØÊ≥®ÂÜåÁïåÈù¢ËøîÂõûÂà∞‰∏ªÁïåÈù¢ÁöÑÊåâÈíÆ‰ø°Âè∑

//--------------------Áî®Êà∑Ê≥®ÂÜåÂπ∂ÁôªÂΩï---------------------------//
    user *us = new user(nullptr);//--------------------------------------ÂàõÈÄ†‰∏Ä‰∏™Áî®Êà∑ÁïåÈù¢
    connect(reg,&registrationWindows::successRegistration,[=](){
        us->UserInitialization(reg->getPhone() );
        us->show();
        beauty();
        reg->close();
        reg->clearLineEdit();
    });

//ËøôÈáåÊòØÁÆ°ÁêÜÂëòÁôªÈôÜÁïåÈù¢
    administratorlogin *admin = new administratorlogin();     //ÂàõÂª∫ÁÆ°ÁêÜÂëòÁôªÂΩïÁïåÈù¢
    connect(ui->administratorBtn,&QPushButton::clicked,[=](){
        admin->show();
        beauty();
        this->hide();
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//---------ÊµãËØïÊúüÈó¥ÊèêÁ§∫ÂØÜÁ†Å--------------//
        {
            QFile *file = new QFile(QString("./data/administor.dat"));
            QDataStream stream(file);
            file->open(QIODevice::ReadOnly);
            int mycode ;
            stream >> mycode;
            file->close();
            QString str = str.number(mycode);
            QMessageBox *box= new QMessageBox();
            box->information(this , QString("ÊèêÁ§∫üêñ") , QString("ÊµãËØïÊúüÈó¥ÔºåÁÆ°ÁêÜÂëòÂØÜÁ†ÅÊòØÔºö")+str+QString("„ÄÇÊÇ®ÂèØ‰ª•Êõ¥ÊîπÂØÜÁ†ÅÔºÅ"),QMessageBox::Ok ,QMessageBox::Ok);
            box->setButtonText(QMessageBox::Ok,QString("ÊàëÁü•ÈÅì‰∫Ü"));
        }
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
    });
    connect(admin ,&administratorlogin::backmain,[=](){
        this->show();
        beauty();
        admin->close();

    });      //----------------------------------------------------------ËøôÈáåÊòØÁÆ°ÁêÜÂëòÁôªÈôÜÁïåÈù¢ËøîÂõûÂáΩÊï∞



//-----------ËøôÊòØÁÆ°ÁêÜÂëòÁôªÂΩïÊìç‰Ωú,ad‰ª£Ë°®ÁÆ°ÁêÜÂëò
     administrator *ad = new administrator();
     connect(admin , &administratorlogin::login,[=](){
            ad->administratorInitialization();
            ad->show();
            beauty();
            admin->close();
        });

     connect(ad,&administrator::backMain,[=](){
         this->show();
         beauty();
         ad->close();
         ad->deletCC();
     });

//-----------‰ªéÁÆ°ÁêÜÂëòÁïåÈù¢ËøîÂõû

//ËøôÈáåÊòØËøõÂÖ•Áî®Êà∑ÁïåÈù¢Ê≥®ÊÑèÊ£ÄÊü•ÂØÜÁ†ÅÔºåÊâãÊú∫Âè∑ÁöÑÊ≠£Á°ÆÊÄß
    connect(ui->enterBtn,&QPushButton::clicked,[=](){ //ËøõÂÖ•Áî®Êà∑ÁïåÈù¢
        if(successlog(  ui->phoneinput->text()  ,  ui->codeinput->text()  ) ){
            us->UserInitialization( ui->phoneinput->text().toLongLong() );
            us->show();
            beauty();
            this->hide();
        }
   });
    connect(us,&user::exitlogin,[=](){               //ÁÇπÂáªËøîÂõûÊåâÈíÆËøîÂõû‰∏ªÁ™óÂè£
        ui->codeinput->setText(QString(""));
        us->deleteCan();
        this->show();
        beauty();
        us->close();

    });




}//---------------------------------------------------------------------------------------------------------ËøôÈáåÊòØÊûÑÈÄ†ÂáΩÊï∞ÁªàÊ≠¢ÁöÑÂè≥Êã¨Âè∑








void MainWindow::InitialCheck(){          ///////ÂáΩÊï∞ÁöÑÂäüËÉΩÂ∞±ÊòØËé∑Âèñ‰ªäÂ§©ÁöÑÊó∂Èó¥ÔºåÁÑ∂ÂêéÂà§Êñ≠ÊòØÂê¶Â≠òÂú®‰ø°ÊÅØÊñá‰ª∂
    aDay = aDay.currentDate();//-----------Ëé∑Âèñ‰ªäÂ§©ÁöÑÊó∂Èó¥

    QDir dir;
    if(!dir.exists(QString("./data/"))){
        dir.mkdir(QString("./data/"));
    }

    {
        QFile *file = new QFile(QString("./data/users.dat"));
        if(!(file->exists())){
            QDataStream stream (file);
            file->open(QIODevice::WriteOnly);
            stream << 0;
            file->close();
        }}           ////////////////////‰∫∫Áâ©Â§ßÂÖ®ÁöÑÂàõÂª∫

    {
        QFile *file = new QFile(QString("./data/administor.dat"));
        if(!(file->exists())){
            QDataStream stream (file);
            file->open(QIODevice::WriteOnly);
            stream << 88888888 << 0 << 0 ;
            file->close();
        }}            //////////////////////ÁÆ°ÁêÜÂëòÁöÑÂàõÂª∫


     {
        QFile *file = new QFile(QString("./data/activities.dat"));
        if(!(file->exists())){
            QDataStream stream (file);
            file->open(QIODevice::WriteOnly);
            stream << 0  ;
            file->close();
        }}            /////////////////////////Ê¥ªÂä®Â§ßÂÖ®ÁöÑÂàõÂª∫





}//-------------------------------------------------------------------------//





#include<QDebug>



///////Ê£ÄÊü•ÊòØÂê¶Ê≠£Á°ÆÁôªÂΩï
bool MainWindow::successlog(QString phone, QString code){           ///
    QFile *file = new QFile(PATH+phone+endPATH);
    if(!file->open(QIODevice::ReadOnly)){
        QMessageBox *box  = new QMessageBox();
        box->critical(this , QString("ÈîôËØØ‚ö†") , QString("Áî®Êà∑Âêç‰∏çÊ≠£Á°ÆÔºÅÔºÅ") ,QMessageBox::Ok ,QMessageBox::Ok);
        box->setButtonText(QMessageBox::Ok,QString("ÈáçËØï"));
        return false;
    }
    long long int myphone ;
    QString str;
    int mycode ;
    QDataStream stream(file);
    stream >> myphone >> str >> mycode;
    if(code.toInt() != mycode){
        QMessageBox *box  = new QMessageBox();
        box->critical(this , QString("ÈîôËØØ‚ö†") , QString("ÂØÜÁ†Å‰∏çÊ≠£Á°ÆÔºÅÔºÅ") ,QMessageBox::Ok ,QMessageBox::Ok);
        box->setButtonText(QMessageBox::Ok,QString("ÈáçËØï"));
        qDebug() << myphone << str << mycode ;
        return false;
    }
    return true;

}



void MainWindow::beauty(){
    QTime dietime = QTime::currentTime().addMSecs(200);
    while(QTime::currentTime()<dietime)
        QCoreApplication::processEvents(QEventLoop::AllEvents,100);
}

MainWindow::~MainWindow()
{
    delete ui;
}

